### PR TITLE
Bulk Export/Import: $export-status Operation support for Delete #112

### PR DESCRIPTION
**Pull request by prb112**
- Clean up BulkData Constants
- Update BulkData Operations
- Refactor the BulkDataClient
- Also implemented, Refine $import, $export and $export-status
operations #55

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

**Comment by prb112** [comment]
Note, I adjusted the points taken on the PR.  (3 in total)

File: fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/cache/BulkDataTenantSpecificCache.java [hunk 1]
```diff
@@ -17,13 +16,11 @@
 import com.ibm.fhir.operation.bulkdata.config.BulkDataConfigUtil;
 
 /**
- * This class implements a cache of BulkData configuration organized by tenantId. Each object stored in the cache will be a
+ * This class implements a cache of BulkData configuration organized by tenantId. Each object stored in the cache will
+ * be a
  * two-level map of SearchParameters organized first by resource type, then by search parameter name.
```
Comment (albertwang-ibm) [inline fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/cache/BulkDataTenantSpecificCache.java]: SearchParameters or jobParameters?

Comment (prb112) [reply fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/cache/BulkDataTenantSpecificCache.java]: funny - we never caught this in the initial implementation. I'll fix. It's actually suppose to say a map of properties related to the bulkdata export.

**Comment by albertwang-ibm** [comment]
The refactoring and the new job delete codes look great for me!!
But this does remind that we need a mechanism to protect the job instance id in the polling location url, otherwise, user will be able to poll and get the exported url of any other user, or delete the request submitted by others.  seems a simple solution is to encrypt the job instance Id, and because the javabatch job id is not reused(always increase if non in-memory db) in server side,  so user will not be able to use his/her old job info to get or deleted other's export.

**Comment by prb112** [comment]
> The refactoring and the new job delete codes look great for me!!
> But this does remind that we need a mechanism to protect the job instance id in the polling location url, otherwise, user will be able to poll and get the exported url of any other user, or delete the request submitted by others. seems a simple solution is to encrypt the job instance Id, and because the javabatch job id is not reused(always increase if non in-memory db) in server side, so user will not be able to use his/her old job info to get or deleted other's export.

Let's sync up on Monday.  I have a good place to put it, and I can explain what I do in verifyTenant.

File: fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/cache/BulkDataTenantSpecificCache.java [hunk 2]
```diff
@@ -17,13 +16,11 @@
 import com.ibm.fhir.operation.bulkdata.config.BulkDataConfigUtil;
 
 /**
- * This class implements a cache of BulkData configuration organized by tenantId. Each object stored in the cache will be a
+ * This class implements a cache of BulkData configuration organized by tenantId. Each object stored in the cache will
```
Comment (prb112) [inline fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/cache/BulkDataTenantSpecificCache.java]:
```suggestion
 * This class implements a cache of BulkData configuration organized by tenantId. Each object stored in the cache is a 
```

File: fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/cache/BulkDataTenantSpecificCache.java [hunk 3]
```diff
@@ -17,13 +16,11 @@
...
+ * be a
```
Comment (prb112) [inline fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/cache/BulkDataTenantSpecificCache.java]:
```suggestion
```

File: fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/cache/BulkDataTenantSpecificCache.java (hunk 1 above)
Comment (prb112) [inline fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/config/cache/BulkDataTenantSpecificCache.java]:
```suggestion
 * a map of properties related to bulkdata operations. 
```

**Pull request merged**